### ISSUE TITLE
Changing the current OpenCl implementation to Vulkan.  [$399]

### ISSUE THREAD

**Issue by matthewv1998** [LuxCoreRender/BlendLuxCore#549]
Hello. With the current, buggy OpenCl implementation, over to Vulkan The goal of this being replace the current OpenCl implementation with a more stable (and more importantly, modern, code base) Further, this is to protect AMD cards within luxcore to keep the engine non-platform exclusive.

**Comment by Dade916** [comment]
This request doesn't make any sense, AMD ROCm a framework (aka a marketing slogan) for the support of the following programming-language choices:

- OpenMP
- HIP
- **OpenCL**
- Python

We already support OpenCL. Just read AMD ROCm web page: https://www.amd.com/en/graphics/servers-solutions-rocm-hpc

Are you also aware the ROCm is supported only on Linux ?

And that it does work only with very few AMD GPUs ? Check: https://github.com/RadeonOpenCompute/ROCm#Hardware-and-Software-Support

Short version: if you want to use ROCm, just install AMD ROCm drivers (i.e. ROCm OpenCL drivers) and run LuxCore using OpenCL devices.

P.S. I close this issue so you can claim the bounty for your self and get back your money.

**Comment by matthewv1998** [comment]
Alright, if that wont work, what about re-opening this Vulkan?

**Comment by Dade916** [comment]
> Alright, if that wont work, what about re-opening this with Vulkan?

Vulkan is a far more interesting and useful possibility. I see Vulkan as a future replacement for OpenCL.

P.S. you are assuming Vulkan AMD drivers work fine, may be or may be not, it is unknown.

**Comment by matthewv1998** [comment]
Alright, i can edit this for vulkan, and have this bounty re-opened, that work?

**Comment by Dade916** [comment]
Yes, it should. If you are not the only backer of the bounty, the other one may not agree with the change of "target" but it will be up to them to speak. And given that the first target wasn't making sense, I assume they will agree.

**Comment by matthewv1998** [comment]
alright, ive edited it but i'll wait for the response.

**Comment by betstick** [comment]
Yes, I support the change to from Rocm to Vulkan at Dade's recommendation.

**Comment by matthewv1998** [comment]
> Yes, it should. If you are not the only backer of the bounty, the other one may not agree with the change of "target" but it will be up to them to speak. And given that the first target wasn't making sense, I assume they will agree.

You have tje green light to open this back up

**Comment by betstick** [comment]
Yes, I support the change to from Rocm to Vulkan at Dade's recommendation.

**Comment by betstick** [comment]
Yes, I support the change to from Rocm to Vulkan at Dade's recommendation.

**Comment by Dade916** [comment]
Note: it is about adding Vulkan support (like OpenCL and CUDA), not replacing OpenCL with Vulkan.

I'm also going to wait for the release AMD ray tracing capable GPUs (and may be Intel), Vulkan has a ray tracing acceleration extension and I assume it will be supported by AMD Vulkan ... but better safe than sorry.

**Comment by Dade916** [comment]
I should transfer this issue to LuxCore repository but I'm hesitant because it could mess with BountySource; I will leave it here for the moment.

**Comment by Ahr0n** [comment]
Hello @Dade916, it looks like the Vulkan implementation could help everyone who is suffering from OCL problems and driver issues. My recent issue with OCL in Luxcore after upgrading OS makes GPU rendering impossible on my computer: https://github.com/LuxCoreRender/LuxCore/issues/468#issuecomment-726884400

The only solution is to make a Vulkan and MoltenVK version of LuxCore for AMD and MacOS users. The Bounty backing is finished so there is a question - is there any hope we will see a Vulkan version of LuxCore soon?

**Comment by matthewv1998** [comment]
@Dade916 hey if you need any testing with a RDNA 2 gpu for this, you can hit me up, i own a 6900xt now.

**Comment by NightHammer1000** [comment]
A am too on a 6900XT now after the 2080ti bit the dust.

I can also test if nessassary.

**Comment by Randrianasulu** [comment]
there is also (experimental, not fast) vulkan ray tracing support for older AMD cards:

'This PR implements ray-tracing for older generations (Navi, Vega, Polaris, etc.)'

https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/12974
